language-support/ts/codegen/src/TsCodeGenMain.hs

```diff
@@ -743,7 +743,7 @@ setupWorkspace optInputPackageJson optOutputDir dependencies = do
     (Just <$> BSL.readFile optInputPackageJson) (const $ pure Nothing)
   packageJson <- case mbBytes of
     Nothing -> pure mempty
-    Just bytes -> case decode bytes :: Maybe PackageJson of
+    Just bytes -> case decode @PackageJson bytes of
```

Comment:
If you use `eiherDecode` you can get an error message on parse failures.

Reply:
Excellent idea. Would be `eitherDecode @PackageJson` then I guess.

Reply:
I'll get another PR going for that.

Reply:
I've had a look at the error messages it produces. They don't look helpful to be honest. For example, with a stray ',' the resulting error message is:
```
Error in $: Failed reading: satisfy. Expecting object value
```
I recommend we do not switch from `decode` to `eitherDecode`.

Reply:
It depends on the error and the kind of code you write. E.g. for a missing key you can often get fairly decent errors. Using things like `withObject`, `<?>` you can improve errors. I wouldn’t use one bad error message as an indicator that not showing the error at all is better.

Reply:
Ok, sounds reasonable. In which case I offer https://github.com/digital-asset/daml/pull/5098 😃